vmdb/app/models/vm_openstack/remote_console.rb

```diff
@@ -0,0 +1,35 @@
+class VmOpenstack
+  module RemoteConsole
+    def console_supported?(type)
+      %w(SPICE VNC).include?(type.upcase)
+    end
+
+    def remote_console_acquire_ticket(console_type, _proxy_miq_server = nil)
+      url = ext_management_system.with_provider_connection(:service => "Compute") do |con|
+        response = con.get_vnc_console(self.ems_ref, 'novnc')
+        return nil if response.body.fetch_path('console', 'type') != 'novnc'
+        response.body.fetch_path('console', 'url')
+      end
+      return nil, url, nil, nil, nil, 'novnc_url', nil
+    end
+
+    def remote_console_acquire_ticket_queue(protocol, userid, proxy_miq_server = nil)
+      task_opts = {
+        :action => "acquiring Instance #{name} #{protocol.to_s.upcase} remote console ticket for user #{userid}",
+        :userid => userid
+      }
+
+      queue_opts = {
+        :class_name  => self.class.name,
+        :instance_id => id,
+        :method_name => 'remote_console_acquire_ticket',
+        :priority    => MiqQueue::HIGH_PRIORITY,
+        :role        => 'ems_operations',  # FIXME
```

Comment:
@blomquisg, @Fryguy : I kept the FIXME there to remind me to ask you if this was the correct role of OpenStack API request processing. Please, check and tell. Thx!